client/src/components/categorical/category/index.tsx

```diff
@@ -114,15 +118,27 @@ class Category extends React.PureComponent<CategoryProps> {
     );
   }
 
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  handleColorChange = () => {
+  handleColorChange = (isColorAccessor: boolean) => {
     // @ts-expect-error ts-migrate(2339) FIXME: Property 'dispatch' does not exist on type 'Readon... Remove this comment to see the full error message
     const { dispatch, metadataField, categoryType } = this.props;
 
-    track(EVENTS.EXPLORER_COLORBY_CATEGORIES_BUTTON_CLICKED, {
-      type: categoryType,
-      category: metadataField,
-    });
+    /**
+     * (thuang): If we're going from `isColorAccessor` being `false` to `true`,
+     * we should track the event!
+     */
+    if (!isColorAccessor) {
```

Comment:
If `isColorAccessor` is `true`, `!isColorAccessor` will be `false`

Reply:
Yeah I'm struggling to make this clearer lol So the idea is that `handleColorChange()` will be called with the category's current `isColorAccessor` state. So if it's `true`, it means that the category is already being used to color the graph, so we will be turning it off. So if it's `false`, it means that the user is about to color the graph using this category, thus us tracking the event!

I just updated the argument name from `isColorAccessor` to `currentIsColorAccessor` - not sure if that helps or not 😆

Reply:
OK I see now. I think I was confused of by the inline comment  lol

> If we're going from `isColorAccessor` being `false` to `true`,
     * we should track the event!